mage_ai/io/base.py

```diff
@@ -170,7 +170,40 @@ def __del__(self):
             print('')
 
 
-class BaseSQL(BaseIO):
+class BaseSQLDatabase(BaseIO):
+    """
+    Base data loader for connecting to a SQL database. This adds `query` method which allows a user
+    to send queries to the databse server.
+    """
+
+    @abstractmethod
+    def query(self, query_string: str, **kwargs) -> None:
```

Comment:
This method it not only querying data, right? `execute` could be a better name.

Reply:
Renamed